core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7433DeadlockTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.it;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.apache.maven.it.util.ResourceExtractor;
+
+public class MavenITmng7433DeadlockTest extends AbstractMavenIntegrationTestCase
+{
+    private static final String PROJECT_PATH = "/mng-7433-deadlock";
+
+    public MavenITmng7433DeadlockTest()
+    {
+        super( "[3.8.4,)" );
```

Comment:
Does this verify that there is a deadlock or your locks avoid it? Because the version says 3.8.4.

Reply:
The test will just hang forever if running with 3.8.5.  It can be run with any other version afaik.

Reply:
Then the range must be fixed for: `(,3.8.4],[3.8.6,)`

Reply:
Fixed